libs/community/langchain_community/vectorstores/yellowbrick.py

```diff
@@ -19,157 +26,374 @@
 
 from langchain_community.docstore.document import Document
 
-logger = logging.getLogger(__name__)
+if TYPE_CHECKING:
+    from psycopg2.extensions import connection as PgConnection
+    from psycopg2.extensions import cursor as PgCursor
 
 
 class Yellowbrick(VectorStore):
     """Yellowbrick as a vector database.
-
     Example:
         .. code-block:: python
             from langchain_community.vectorstores import Yellowbrick
             from langchain_community.embeddings.openai import OpenAIEmbeddings
             ...
     """
 
+    class IndexType(str, enum.Enum):
+        """Enumerator for the supported Index types within Yellowbrick."""
+
+        NONE = "none"
+        LSH = "lsh"
+
+    class IndexParams:
+        """Parameters for configuring a Yellowbrick index."""
+
+        def __init__(
+            self,
+            index_type: Optional["Yellowbrick.IndexType"] = None,
+            params: Optional[Dict[str, Any]] = None,
+        ):
+            if index_type is None:
+                index_type = Yellowbrick.IndexType.NONE
+            self.index_type = index_type
+            self.params = params or {}
+
+        def get_param(self, key: str, default: Any = None) -> Any:
+            return self.params.get(key, default)
+
     def __init__(
         self,
         embedding: Embeddings,
         connection_string: str,
         table: str,
+        *,
+        schema: Optional[str] = None,
+        logger: Optional[logging.Logger] = None,
+        seed: Optional[float] = 0.42,
+        drop: Optional[bool] = False,
```

Comment:
nit these probably dont need to be typed as Optional

Reply:
Makes sense. Changed. Thanks!